Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,187 @@
+         */
+        render: function () {
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Toggles the displayed value.
+         *
+         * @method toggleDisplayed
+         * @public
+         */
+        toggleDisplayed: function () {
+            this._set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _uiToggleUserMenu
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiToggleUserMenu: function (event) {
+            var userMenu = this.get('container');
+
+            if (event.newVal) {
+                userMenu.removeClass(CLASS_HIDDEN);
+            } else {
+                userMenu.addClass(CLASS_HIDDEN);
+            }
+        },
+
+        /**
+         * Set the click outside event listener
+         *
+         * @method _setClickOutsideUserMenuHandler
+         * @protected
+         * @param event {Object} event facade
+         */
+        _setClickOutsideUserMenuHandler: function (event) {
+            if (event.newVal) {
+                this._clickOutsideUserMenuHandler = this.get('container').on('clickoutside', this._hideMenuOnClickOutside, this);
+            } else {
+                this._clickOutsideUserMenuHandler.detach();
+            }
+
+            return this;
+        },
+
+        /**
+         * Checks if clicked outside the user menu container and hides a user menu if it is visible.
+         *
+         * @method _hideMenuOnClickOutside
+         * @protected
+         */
+        _hideMenuOnClickOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
+                this.toggleDisplayed();
+            }
+        },
+
+        /**
+         * Sets active flag to the item view
+         *
+         * @method _setActiveViews
+         * @protected
+         */
+        _setActiveViews: function () {
+            var items = this.get('items');
+
+            items.forEach(function (item) {
+                item.set('active', true);
+            });
+        },
+
+        /**
+         * Renders menu items views
+         *
+         * @method _renderMenuItems
+         * @protected
+         */
+        _renderMenuItems: function () {
+            var items = this.get('items');
+
+            items.sort(function (a, b) {
+                return b.get('priority') - a.get('priority');
+            });
+
+            items.forEach(Y.bind(this._addMenuItem, this));
+        },
+
+        /**
+         * Adds item to the menu
+         *
+         * @method _addMenuItem
+         * @protected
+         * @param view {Object}
+         */
+        _addMenuItem: function (view) {
+            var menuItemsContainer = this.get('container');
+
+            view.addTarget(this);
```

Comment:
it seems like I was not clear enough in https://github.com/ezsystems/PlatformUIBundle/pull/513#discussion_r53769841 :(

Calling `addTarget` at render time is not correct because you might expect some events from child view before the view is rendered and this is absolutely not related to the render logic.

So my initial comment was about separating `addTarget` from the rendering logic. Basically, `addTarget` should be called when a view is added an item of the `UserMenuView`. To do that, in my initial comment, I was suggesting that you add a public method called `addMenuItem` that would take a view into parameter and add it to the `items` attribute and after doing that would call `addTarget`.
Then when rendering `UserMenuView` you would just iterate over the items, render them and add the container to the UserMenuView's container.